In [244]:
import numpy as np
import matplotlib.pyplot as plt

In [1]:
import pandas as pd

stn_ids = pd.read_fwf('http://noaa-ghcn-pds.s3.amazonaws.com/ghcnd-stations.txt', header=None, infer_nrows=1000)
stn_ids.columns = ['ID','LAT','LON','ELEV','UKN','NAME','GSN','WBAN']
stn_ids

ID      LAT      LON    ELEV  UKN                   NAME  \
0       ACW00011604  17.1167 -61.7833    10.1  NaN  ST JOHNS COOLIDGE FLD   
1       ACW00011647  17.1333 -61.7833    19.2  NaN               ST JOHNS   
2       AE000041196  25.3330  55.5170    34.0  NaN    SHARJAH INTER. AIRP   
3       AEM00041194  25.2550  55.3640    10.4  NaN             DUBAI INTL   
4       AEM00041217  24.4330  54.6510    26.8  NaN         ABU DHABI INTL   
...             ...      ...      ...     ...  ...                    ...   
129653  ZI000067969 -21.0500  29.3670   861.0  NaN         WEST NICHOLSON   
129654  ZI000067975 -20.0670  30.8670  1095.0  NaN               MASVINGO   
129655  ZI000067977 -21.0170  31.5830   430.0  NaN          BUFFALO RANGE   
129656  ZI000067983 -20.2000  32.6160  1132.0  NaN               CHIPINGE   
129657  ZI000067991 -22.2170  30.0000   457.0  NaN             BEITBRIDGE   

        GSN     WBAN  
0       NaN      NaN  
1       NaN      NaN  
2       GSN  41196.0  
3       NaN  41194.0  
4       NaN  41217.0  
...     ...      ...  
129653  NaN  67969.0  
129654  NaN  67975.0  
129655  NaN  67977.0  
129656  GSN  67983.0  
129657  NaN  67991.0  

[129658 rows x 8 columns]

In [2]:
periods = pd.read_fwf('http://noaa-ghcn-pds.s3.amazonaws.com/ghcnd-inventory.txt', header=None, infer_nrows=1000)
periods.columns = ['ID','LAT','LON','ELEM','TiMIN','TiMAX']
periods

ID      LAT      LON  ELEM  TiMIN  TiMAX
0       ACW00011604  17.1167 -61.7833  TMAX   1949   1949
1       ACW00011604  17.1167 -61.7833  TMIN   1949   1949
2       ACW00011604  17.1167 -61.7833  PRCP   1949   1949
3       ACW00011604  17.1167 -61.7833  SNOW   1949   1949
4       ACW00011604  17.1167 -61.7833  SNWD   1949   1949
...             ...      ...      ...   ...    ...    ...
767508  ZI000067983 -20.2000  32.6160  PRCP   1951   2025
767509  ZI000067983 -20.2000  32.6160  TAVG   1962   2025
767510  ZI000067991 -22.2170  30.0000  TMAX   1951   1990
767511  ZI000067991 -22.2170  30.0000  TMIN   1951   1990
767512  ZI000067991 -22.2170  30.0000  PRCP   1951   1990

[767513 rows x 6 columns]

In [3]:
merged_stns = pd.merge(stn_ids,periods,how='left',left_on='ID',right_on='ID')
merged_stns

ID    LAT_x    LON_x    ELEV  UKN                   NAME  \
0       ACW00011604  17.1167 -61.7833    10.1  NaN  ST JOHNS COOLIDGE FLD   
1       ACW00011604  17.1167 -61.7833    10.1  NaN  ST JOHNS COOLIDGE FLD   
2       ACW00011604  17.1167 -61.7833    10.1  NaN  ST JOHNS COOLIDGE FLD   
3       ACW00011604  17.1167 -61.7833    10.1  NaN  ST JOHNS COOLIDGE FLD   
4       ACW00011604  17.1167 -61.7833    10.1  NaN  ST JOHNS COOLIDGE FLD   
...             ...      ...      ...     ...  ...                    ...   
767572  ZI000067983 -20.2000  32.6160  1132.0  NaN               CHIPINGE   
767573  ZI000067983 -20.2000  32.6160  1132.0  NaN               CHIPINGE   
767574  ZI000067991 -22.2170  30.0000   457.0  NaN             BEITBRIDGE   
767575  ZI000067991 -22.2170  30.0000   457.0  NaN             BEITBRIDGE   
767576  ZI000067991 -22.2170  30.0000   457.0  NaN             BEITBRIDGE   

        GSN     WBAN    LAT_y    LON_y  ELEM   TiMIN   TiMAX  
0       NaN      NaN  17.1167 -61.7833  TMAX  1949.0  1949.0  
1       NaN      NaN  17.1167 -61.7833  TMIN  1949.0  1949.0  
2       NaN      NaN  17.1167 -61.7833  PRCP  1949.0  1949.0  
3       NaN      NaN  17.1167 -61.7833  SNOW  1949.0  1949.0  
4       NaN      NaN  17.1167 -61.7833  SNWD  1949.0  1949.0  
...     ...      ...      ...      ...   ...     ...     ...  
767572  GSN  67983.0 -20.2000  32.6160  PRCP  1951.0  2025.0  
767573  GSN  67983.0 -20.2000  32.6160  TAVG  1962.0  2025.0  
767574  NaN  67991.0 -22.2170  30.0000  TMAX  1951.0  1990.0  
767575  NaN  67991.0 -22.2170  30.0000  TMIN  1951.0  1990.0  
767576  NaN  67991.0 -22.2170  30.0000  PRCP  1951.0  1990.0  

[767577 rows x 13 columns]

In [11]:
pc_data = merged_stns.iloc[np.where(merged_stns['ID']=='USC00087205')[0]]
pc_data

ID    LAT_x    LON_x  ELEV UKN        NAME  GSN  WBAN  \
469181  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469182  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469183  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469184  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469185  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469186  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469187  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469188  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469189  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469190  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469191  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469192  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469193  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469194  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469195  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   
469196  USC00087205  28.0208 -82.1392  33.2  FL  PLANT CITY  NaN   NaN   

          LAT_y    LON_y  ELEM   TiMIN   TiMAX  
469181  28.0208 -82.1392  TMAX  1892.0  2025.0  
469182  28.0208 -82.1392  TMIN  1892.0  2025.0  
469183  28.0208 -82.1392  TOBS  1901.0  2025.0  
469184  28.0208 -82.1392  PRCP  1893.0  2025.0  
469185  28.0208 -82.1392  SNOW  1901.0  2025.0  
469186  28.0208 -82.1392  SNWD  1901.0  2025.0  
469187  28.0208 -82.1392  DAPR  1949.0  2004.0  
469188  28.0208 -82.1392  MDPR  1949.0  2008.0  
469189  28.0208 -82.1392  WT01  2008.0  2014.0  
469190  28.0208 -82.1392  WT03  1951.0  2025.0  
469191  28.0208 -82.1392  WT04  2009.0  2009.0  
469192  28.0208 -82.1392  WT06  2008.0  2008.0  
469193  28.0208 -82.1392  WT08  1938.0  1938.0  
469194  28.0208 -82.1392  WT11  1949.0  2022.0  
469195  28.0208 -82.1392  WT14  1950.0  1950.0  
469196  28.0208 -82.1392  WT16  1903.0  1923.0

In [14]:
pc_df = pd.read_csv(
...      "s3://noaa-ghcn-pds/csv/by_station/USC00087205.csv",
...      storage_options={"anon": True},  # passed to `s3fs.S3FileSystem`
         dtype={'Q_FLAG': 'object', 'M_FLAG': 'object'},
         parse_dates=['DATE']
... ).set_index('DATE')

pc_df

/tmp/ipykernel_3674305/272402385.py:1: DtypeWarning: Columns (6) have mixed types. Specify dtype option on import or set low_memory=False.
  pc_df = pd.read_csv(


ID ELEMENT  DATA_VALUE M_FLAG Q_FLAG S_FLAG  OBS_TIME
DATE                                                                      
1892-09-01  USC00087205    TMAX         322    NaN    NaN      6       NaN
1892-09-02  USC00087205    TMAX         317    NaN    NaN      6       NaN
1892-09-03  USC00087205    TMAX         317    NaN    NaN      6       NaN
1892-09-04  USC00087205    TMAX         322    NaN    NaN      6       NaN
1892-09-05  USC00087205    TMAX         333    NaN    NaN      6       NaN
...                 ...     ...         ...    ...    ...    ...       ...
2025-12-04  USC00087205    PRCP           0    NaN    NaN      H    1600.0
2025-12-05  USC00087205    PRCP           0    NaN    NaN      H    1600.0
2025-12-06  USC00087205    PRCP           0    NaN    NaN      H    1600.0
2025-12-07  USC00087205    PRCP         185    NaN    NaN      H    1600.0
2025-12-08  USC00087205    PRCP         406    NaN    NaN      H    1600.0

[224223 rows x 7 columns]

In [15]:
pc_tmin = pc_df.loc[pc_df['ELEMENT'] == 'TMIN']
pc_tmin

ID ELEMENT  DATA_VALUE M_FLAG Q_FLAG S_FLAG  OBS_TIME
DATE                                                                      
1892-09-01  USC00087205    TMIN         206    NaN    NaN      6       NaN
1892-09-02  USC00087205    TMIN         206    NaN    NaN      6       NaN
1892-09-03  USC00087205    TMIN         211    NaN    NaN      6       NaN
1892-09-04  USC00087205    TMIN         217    NaN    NaN      6       NaN
1892-09-05  USC00087205    TMIN         211    NaN    NaN      6       NaN
...                 ...     ...         ...    ...    ...    ...       ...
2025-12-04  USC00087205    TMIN         122    NaN    NaN      H    1600.0
2025-12-05  USC00087205    TMIN         178    NaN    NaN      H    1600.0
2025-12-06  USC00087205    TMIN         206    NaN    NaN      H    1600.0
2025-12-07  USC00087205    TMIN         156    NaN    NaN      H    1600.0
2025-12-08  USC00087205    TMIN         172    NaN    NaN      H    1600.0

[45414 rows x 7 columns]

# 1)
Strawberries are planted around October 1 and ready for harvest by the end of January. 
 - October 1 - January 30

What is the MEAN risk of frost and freeze?
- Defined as the mean number of days per month over the period 1991-2020
- Frost temps <= 32 F = 0 C
- Freeze temps <= 28 F = -2.2 C

In [36]:
pc_tmin_yrs = pc_tmin.iloc[np.where((pc_tmin.index.year>=1991)&(pc_tmin.index.year<=2020))[0]]
pc_tmin_sub = pc_tmin_yrs.iloc[np.where((pc_tmin_yrs.index.month>=10)|(pc_tmin_yrs.index.month==1))[0]]
pc_tmin_sub['Year'] = pc_tmin_sub.index.year
pc_tmin_sub['Month'] = pc_tmin_sub.index.month
pc_tmin_sub['ym'] = list(zip(pc_tmin_sub.index.year, pc_tmin_sub.index.month))
pc_tmin_sub['Tmin_corr'] = pc_tmin_sub['DATA_VALUE']/10
pc_tmin_sub

/tmp/ipykernel_3674305/3595140445.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  pc_tmin_sub['Year'] = pc_tmin_sub.index.year
/tmp/ipykernel_3674305/3595140445.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  pc_tmin_sub['Month'] = pc_tmin_sub.index.month
/tmp/ipykernel_3674305/3595140445.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata

ID ELEMENT  DATA_VALUE M_FLAG Q_FLAG S_FLAG  OBS_TIME  \
DATE                                                                         
1991-01-01  USC00087205    TMIN         167    NaN    NaN      0    1800.0   
1991-01-02  USC00087205    TMIN         183    NaN    NaN      0    1800.0   
1991-01-03  USC00087205    TMIN         178    NaN    NaN      0    1800.0   
1991-01-04  USC00087205    TMIN         178    NaN    NaN      0    1800.0   
1991-01-05  USC00087205    TMIN         172    NaN    NaN      0    1800.0   
...                 ...     ...         ...    ...    ...    ...       ...   
2020-12-27  USC00087205    TMIN          11    NaN    NaN      7    1600.0   
2020-12-28  USC00087205    TMIN          89    NaN    NaN      7    1600.0   
2020-12-29  USC00087205    TMIN         117    NaN    NaN      7    1600.0   
2020-12-30  USC00087205    TMIN         128    NaN    NaN      7    1600.0   
2020-12-31  USC00087205    TMIN         189    NaN    NaN      7    1600.0   

            Year  Month          ym  Tmin_corr  
DATE                                            
1991-01-01  1991      1   (1991, 1)       16.7  
1991-01-02  1991      1   (1991, 1)       18.3  
1991-01-03  1991      1   (1991, 1)       17.8  
1991-01-04  1991      1   (1991, 1)       17.8  
1991-01-05  1991      1   (1991, 1)       17.2  
...          ...    ...         ...        ...  
2020-12-27  2020     12  (2020, 12)        1.1  
2020-12-28  2020     12  (2020, 12)        8.9  
2020-12-29  2020     12  (2020, 12)       11.7  
2020-12-30  2020     12  (2020, 12)       12.8  
2020-12-31  2020     12  (2020, 12)       18.9  

[3617 rows x 11 columns]

In [73]:
frost_df = pc_tmin_sub.iloc[np.where(pc_tmin_sub['Tmin_corr']<=0)[0]]
frost_ym_count = frost_df.groupby(by=['Year', 'Month']).count()['ID']
print(frost_ym_count)
frost_risk = frost_ym_count.groupby(level='Month').mean()
frost_risk

Year  Month
1992  1        3
1993  12       1
1995  12       4
1996  1        2
1997  1        2
1999  1        3
      12       1
2000  1        1
      11       1
      12       2
2001  1        9
2002  1        4
2003  1        4
      12       1
2004  1        1
2005  1        2
2008  1        1
2009  1        4
2010  1        8
      12       8
2011  1        2
2012  1        2
2013  12       1
2014  1        1
2016  1        1
2017  1        1
2018  1        4
2020  1        1
Name: ID, dtype: int64


Month
1     2.800000
11    1.000000
12    2.571429
Name: ID, dtype: float64

In [75]:
frost_risk.iloc[1]

np.float64(1.0)

In [80]:
print('The frost risk is \n' \
'0 in October,\n'
'%f in November,\n'
'%f in December,\n'
'and %f in January ' \
'\n based on the mean number of frost days per month from 1990-2020'%(frost_risk.iloc[1], frost_risk.iloc[2], frost_risk.iloc[0]))

The frost risk is 
0 in October,
1.000000 in November,
2.571429 in December,
and 2.800000 in January 
 based on the mean number of frost days per month from 1990-2020


In [81]:
freeze_df = pc_tmin_sub.iloc[np.where(pc_tmin_sub['Tmin_corr']<=-2.2)[0]]
freeze_ym_count = freeze_df.groupby(by=['Year', 'Month']).count()['ID']
print(freeze_ym_count)
freeze_risk = freeze_ym_count.groupby(level='Month').mean()
freeze_risk

Year  Month
1992  1        1
1995  12       3
1996  1        1
1997  1        1
1999  1        3
2001  1        1
2002  1        1
2003  1        1
2008  1        1
2009  1        1
2010  1        5
      12       3
2014  1        1
2018  1        2
Name: ID, dtype: int64


Month
1     1.583333
12    3.000000
Name: ID, dtype: float64

In [84]:
print('The freeze risk is \n' \
'0 in October,\n'
'0 in November,\n'
'%f in December,\n'
'and %f in January ' \
'\n based on the mean number of freeze days per month from 1990-2020'%(freeze_risk.iloc[1], freeze_risk.iloc[0]))

The freeze risk is 
0 in October,
0 in November,
3.000000 in December,
and 1.583333 in January 
 based on the mean number of freeze days per month from 1990-2020


# 2) Which SST forcing region is most related to cold conditions in central Florida?

Using the temperature anomalies computed in the file, determine which ENSO index (NINO1+2, NINO3, NINO4, and NINO3.4) is best correlated (i.e., has the highest absolute value of Pearson's correlation coefficient) with the number of days per month < 28 degrees F.

In [86]:
import fsspec

In [87]:
# URL of the file
url = "https://www.cpc.ncep.noaa.gov/data/indices/sstoi.indices"

In [235]:
df = pd.read_fwf(url, header=None)
good_df = df.iloc[:,1:5]
df_enso = pd.DataFrame(good_df.values[1:].astype(float), columns = ["ANOM12","NINO3","ANOM3", "NINO4"])
df_enso

ANOM12  NINO3  ANOM3  NINO4
0     -0.24  25.84   0.17  28.01
1     -0.72  26.26  -0.11  27.99
2     -1.38  26.92  -0.25  28.18
3     -1.16  27.52  -0.05  28.61
4     -0.62  27.70   0.49  29.19
..      ...    ...    ...    ...
522    0.46  25.92   0.04  28.84
523    0.23  24.97  -0.24  28.63
524   -0.18  24.60  -0.41  28.41
525   -0.04  24.74  -0.35  28.36
526   -0.34  24.60  -0.60  28.15

[527 rows x 4 columns]

In [236]:
df

0      1      2      3      4              5      6
0     YR MON  NINO1+2   ANOM  NINO3   ANOM  NINO4   ANOM NINO3.4   ANOM
1    1982   1   24.28  -0.24  25.84   0.17  28.01  -0.21   26.65   0.08
2    1982   2   25.38  -0.72  26.26  -0.11  27.99  -0.11   26.54  -0.20
3    1982   3   25.22  -1.38  26.92  -0.25  28.18  -0.05   27.09  -0.14
4    1982   4   24.57  -1.16  27.52  -0.05  28.61   0.10   27.83   0.02
..                ...    ...    ...    ...    ...            ...    ...
523  2025   7   22.29   0.46  25.92   0.04  28.84   0.05   27.24  -0.06
524  2025   8   21.09   0.23  24.97  -0.24  28.63  -0.06   26.58  -0.33
525  2025   9   20.40  -0.18  24.60  -0.41  28.41  -0.27   26.32  -0.44
526  2025  10   20.83  -0.04  24.74  -0.35  28.36  -0.33   26.29  -0.48
527  2025  11   21.28  -0.34  24.60  -0.60  28.15  -0.52   26.14  -0.68

[528 rows x 7 columns]

In [237]:
yr = []
mn = []
nino12 = []
anom4 = []
nino34 = []
for n in range(1,len(df[0])):
    yr.append(int(df[0].iloc[n][0:4]))
    mn.append(int(df[0].iloc[n][6:8]))
    nino12.append(float(df[0].iloc[n][-5::]))

    anom4.append(float(df[5][n][0:5]))
    nino34.append(float(df[5][n][-5::]))

In [238]:
df_enso['Year'] = yr
df_enso['Month'] = mn
df_enso["NINO12"] = nino12
df_enso["ANOM4"] = anom4
df_enso["NINO34"] = nino34
df_enso["ANOM34"] = df[6][1::].values.astype(float)
df_enso

ANOM12  NINO3  ANOM3  NINO4  Year  Month  NINO12  ANOM4  NINO34  ANOM34
0     -0.24  25.84   0.17  28.01  1982      1   24.28  -0.21   26.65    0.08
1     -0.72  26.26  -0.11  27.99  1982      2   25.38  -0.11   26.54   -0.20
2     -1.38  26.92  -0.25  28.18  1982      3   25.22  -0.05   27.09   -0.14
3     -1.16  27.52  -0.05  28.61  1982      4   24.57   0.10   27.83    0.02
4     -0.62  27.70   0.49  29.19  1982      5   24.00   0.40   28.37    0.49
..      ...    ...    ...    ...   ...    ...     ...    ...     ...     ...
522    0.46  25.92   0.04  28.84  2025      7   22.29   0.05   27.24   -0.06
523    0.23  24.97  -0.24  28.63  2025      8   21.09  -0.06   26.58   -0.33
524   -0.18  24.60  -0.41  28.41  2025      9   20.40  -0.27   26.32   -0.44
525   -0.04  24.74  -0.35  28.36  2025     10   20.83  -0.33   26.29   -0.48
526   -0.34  24.60  -0.60  28.15  2025     11   21.28  -0.52   26.14   -0.68

[527 rows x 10 columns]

In [239]:
anom_df = df_enso[['ANOM12','ANOM3','ANOM4', 'ANOM34', 'Year', 'Month']]
anom_df_yrsub = anom_df.iloc[np.where((anom_df['Year']>=1990)&(anom_df['Year']<=2020))[0]]
anom_df_sub = anom_df_yrsub.iloc[np.where((anom_df_yrsub['Month']>=10)|(anom_df_yrsub['Month']==1))[0]]
anom_df_sub

ANOM12  ANOM3  ANOM4  ANOM34  Year  Month
96    -0.60  -0.40  -0.02   -0.11  1990      1
105   -0.79  -0.17   0.28    0.18  1990     10
106   -0.90  -0.36   0.15   -0.09  1990     11
107   -0.72  -0.08   0.39    0.22  1990     12
108   -0.78  -0.05   0.40    0.33  1991      1
..      ...    ...    ...     ...   ...    ...
455    0.27   0.29   0.85    0.44  2019     12
456   -0.03   0.22   0.94    0.52  2020      1
465   -0.76  -0.90  -0.71   -1.13  2020     10
466   -0.67  -1.03  -0.70   -1.23  2020     11
467   -0.86  -0.70  -0.85   -0.99  2020     12

[124 rows x 6 columns]

In [256]:
freeze_ym_count

Year  Month
1992  1        1
1995  12       3
1996  1        1
1997  1        1
1999  1        3
2001  1        1
2002  1        1
2003  1        1
2008  1        1
2009  1        1
2010  1        5
      12       3
2014  1        1
2018  1        2
Name: ID, dtype: int64

In [264]:
anom_df_avg = anom_df_sub.groupby(by=['Year','Month']).mean()
anom_df_avg

ANOM12  ANOM3  ANOM4  ANOM34
Year Month                              
1990 1       -0.60  -0.40  -0.02   -0.11
     10      -0.79  -0.17   0.28    0.18
     11      -0.90  -0.36   0.15   -0.09
     12      -0.72  -0.08   0.39    0.22
1991 1       -0.78  -0.05   0.40    0.33
...            ...    ...    ...     ...
2019 12       0.27   0.29   0.85    0.44
2020 1       -0.03   0.22   0.94    0.52
     10      -0.76  -0.90  -0.71   -1.13
     11      -0.67  -1.03  -0.70   -1.23
     12      -0.86  -0.70  -0.85   -0.99

[124 rows x 4 columns]

In [265]:
merged_df = pd.merge(freeze_ym_count,anom_df_avg, left_index=True, right_index=True, how='inner')
merged_df=merged_df.rename(columns={'ID': 'Freeze Days'})
merged_df

Freeze Days  ANOM12  ANOM3  ANOM4  ANOM34
Year Month                                           
1992 1                1    0.07   1.18   0.58    1.67
1995 12               3   -0.86  -0.96  -0.48   -0.91
1996 1                1   -0.69  -0.71  -0.48   -0.83
1997 1                1   -0.93  -0.98  -0.10   -0.61
1999 1                3   -0.69  -1.15  -1.67   -1.58
2001 1                1   -0.37  -0.43  -0.91   -0.76
2002 1                1   -0.58  -0.65   0.38   -0.20
2003 1                1   -0.05   0.67   0.64    0.88
2008 1                1   -0.26  -1.35  -1.44   -1.59
2009 1                1   -0.16  -0.35  -0.84   -0.80
2010 1                5    0.00   1.06   1.16    1.53
     12               3   -1.20  -1.29  -1.14   -1.21
2014 1                1    0.19  -0.20  -0.32   -0.49
2018 1                2   -0.74  -1.09  -0.20   -0.70

In [266]:
merged_df.corr()

Freeze Days    ANOM12     ANOM3     ANOM4    ANOM34
Freeze Days     1.000000 -0.190068  0.091488  0.122470  0.143197
ANOM12         -0.190068  1.000000  0.754584  0.420404  0.591427
ANOM3           0.091488  0.754584  1.000000  0.784927  0.949753
ANOM4           0.122470  0.420404  0.784927  1.000000  0.910583
ANOM34          0.143197  0.591427  0.949753  0.910583  1.000000

The above correlation matrix shows that the ENSO anomolies most strongly related to freeze days in Plant City are the anomolies associated with ENSO1+2 with a correlation of -0.19.